test/attitude.cpp

```diff
@@ -107,8 +107,9 @@ int main()
     double rad2deg = 180.0 / M_PI;
 
     // euler dcm round trip check
-    for (int roll = -90; roll <= 90; roll += 90) {
-        for (int pitch = -90; pitch <= 90; pitch += 90) {
+    // volatile to prevent loop unrolling and signed overflow warning as error
+    for (volatile int roll = -90; roll <= 90; roll += 90) {
+        for (volatile int pitch = -90; pitch <= 90; pitch += 90) {
```

Comment:
What is the exact error that you get? Is it just a GCC anomaly? `make check` runs through fine for me with GCC 7.2.1

Reply:
```
maetugr @ ubuntu ~/Firmware/src/lib/matrix (normalization-with-zero-check) ▶ make check
Scanning dependencies of target attitude
[  2%] Building CXX object test/CMakeFiles/attitude.dir/attitude.cpp.o
/home/maetugr/Firmware/src/lib/matrix/test/attitude.cpp: In function ‘int main()’:
/home/maetugr/Firmware/src/lib/matrix/test/attitude.cpp:6:5: error: assuming signed overflow does not occur when changing X +- C1 cmp C2 to X cmp C2 -+ C1 [-Werror=strict-overflow]
 int main()
     ^
/home/maetugr/Firmware/src/lib/matrix/test/attitude.cpp:6:5: error: assuming signed overflow does not occur when changing X +- C1 cmp C2 to X cmp C2 -+ C1 [-Werror=strict-overflow]
cc1plus: all warnings being treated as errors
test/CMakeFiles/attitude.dir/build.make:62: recipe for target 'test/CMakeFiles/attitude.dir/attitude.cpp.o' failed
make[3]: *** [test/CMakeFiles/attitude.dir/attitude.cpp.o] Error 1
CMakeFiles/Makefile2:822: recipe for target 'test/CMakeFiles/attitude.dir/all' failed
make[2]: *** [test/CMakeFiles/attitude.dir/all] Error 2
CMakeFiles/Makefile2:74: recipe for target 'CMakeFiles/check.dir/rule' failed
make[1]: *** [CMakeFiles/check.dir/rule] Error 2
Makefile:151: recipe for target 'check' failed
make: *** [check] Error 2
```
Let's put this problem in a separate issue/pr then because it has nothing to do with the normalization and if it works without error on other systems/newer GCC the solution might not be right one.

Reply:
I dropped this commit.

Reply:
Ok, it's just that GCC cannot determine whether an int overflow can occur or not. But I don't see that it can, so adding the volatile with a clear comment is fine for me.